File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left: int = 0
        right: int = len(numbers) - 1
        while left < right:
            temp: int = numbers[left] + numbers[right]
            if temp == target:
                return [left + 1, right + 1]
            elif temp > target:
                right -= 1
            else:
                left += 1
        return []

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {  
    public boolean isSubtree(TreeNode root, TreeNode subRoot) {
        if(root == null) return false;
        boolean b = false;
        if(root.val == subRoot.val) {
           b = isSame(root, subRoot);
        };
        if(b) return true;
        return isSubtree(root.left, subRoot) || isSubtree(root.right, subRoot);
    }

    private boolean isSame(TreeNode t1, TreeNode t2){
        if(t1 == null || t2 == null) return t1 == t2;
        if(t1.val != t2.val) return false;
        return isSame(t1.left, t2.left) && isSame(t1.right, t2.right); 
    }
}
